Doc/library/socket.rst

```diff
@@ -277,6 +277,283 @@ Constants
+   is reached. If the socket is closed due to the program exiting, it
+   will linger in the background.
+
+   The value for this option is a struct on most operating systems.
+   Consult your OS documentation for the struct's details.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_OOBINLINE
+
+   Allows receiving of out of band data in the normal stream.
+
+   .. seealso::
+
+      `Out-of-band data <https://www.ibm.com/support/knowledgecenter/en/ssw_i5_54/rzab6/coobd.htm>`_
+      for an in depth explanation.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_REUSEPORT
+
+..   Availability: Linux >= 3.9 (for TCP/UDP)
+
+.. data:: SO_SNDBUF
+          SO_RCVBUF
+
+   Sets the size of the sending and receiving buffers for this socket in
+   bytes. Most operating systems impose an upper limit on the size of
+   these buffers.
+
+..   Availability: Linux, Windows
+
+.. data::
+          SO_RCVLOWAT
+          SO_SNDLOWAT
+
+   SO_RCVLOWAT sets the minimum number of bytes that must be present in
+   the socket's internal receive buffer before they are passed on to able
+   read call. SO_SNDLOWAT similiary sets the minimum bytes before data is
+   sent from the send buffer to the socket protocol.
+
+   SO_SNDLOWAT is read-only on Linux and SO_RCVLOWAT is read-only on
+   Linux versions below 2.4.
+
+   Both these values default to 1.
+
+..   Availability: Linux, Windows `(non functional) <https://msdn.microsoft.com/en-us/library/windows/desktop/ms740532(v=vs.85).aspx>`_
+
+.. data:: SO_RCVTIMEO
+          SO_SNDTIMEO
+
+   Specifies the amount of time send and receive calls for this socket willl
+   block before timing out. The default timeout of zero means that operations
+   will never time out. On Linux this is a `struct timeval`, on Windows this
+   is the time in milliseconds.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_ERROR
+
+   Gets and resets the pending socket error.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_TYPE
+
+   Gets the type of the socket. This type corresponds to the type as
+   defined in the :func:`socket.socket` function. Examples include
+   :const:`SOCK_STREAM` and :const:`SOCK_DGRAM`.
+
+..   Availability: Linux, Windows
+
+.. data:: SO_SETFIB
+
+   Sets the FIB (routing table) for the socket.
+
+   Availability: FreeBSD >= 7.1
+
+   .. versionadded:: 3.1
+
+.. data:: SO_PASSCRED
+          SO_PEERCRED
+
+   Allows for the passing of SCM credentials over unix sockets.
+   See the end of :func:`socket.recvmsg` for details.
+
+   .. versionadded:: 3.3
+
+..   Availability: Linux
+
+.. data:: SO_PASSSEC
+          SO_PEERSEC
+
+   Allows for the passing of security contexts over unix sockets.
+
+   .. versionadded:: 3.6
+
+..   Availability: Linux >= 2.6.13
+
+.. data:: SO_BINDTODEVICE
+
+   Binds a socket to a particular network interface device like "eth0",
+   When bound, only packets received from that particular device are
+   processsed by the socket.
+
+   .. versionadded:: 3.1
```

Comment:
Oh I came to the 3.1 number by going to https://github.com/python/cpython/blob/master/Modules/socketmodule.c

Clicking the blame button and checking the tag in the commit. The tag says 3.1 but it looks like that is incorrect, since the [README at the time](https://github.com/python/cpython/blob/cb98bed2992022d85d456c146a956e248be04000/README) says: `This is Python version 3.3 alpha 0`